drivers/serial/uart_sam0.c

```diff
@@ -775,7 +775,7 @@ static int uart_sam0_irq_rx_ready(const struct device *dev)
 {
 	SercomUsart * const regs = DEV_CFG(dev)->regs;
 
-	return regs->INTFLAG.bit.RXC != 0;
+	return (regs->INTFLAG.bit.RXC != 0) && (regs->INTENSET.bit.RXC != 0);
```

Comment:
If I remember correctly, that is not right at least for rx_ready, see my comment for CDC ACM, https://github.com/zephyrproject-rtos/zephyr/pull/38155#discussion_r699273701

Reply:
Makes sense.  When I get a chance, I'll update to not change this line later today.

Reply:
Tested without this change and TX transmission works as expected and RX is not corrupted.
I also verified that when transmitting writes to DATA, it does not trigger a RXC INTFLAG interrupt.